packages/astro/components/ViewTransitions.astro

```diff
@@ -168,13 +168,18 @@ const { fallback = 'animate' } = Astro.props as Props;
 
 			// Trigger the animations
 			document.documentElement.dataset.astroTransitionFallback = 'old';
-			doc.documentElement.dataset.astroTransitionFallback = 'new';
+			const fallbackSwap = () => {
+				removeEventListener('animationend', fallbackSwap);
+				clearTimeout(timeout);
+				swap();
+				document.documentElement.dataset.astroTransitionFallback = 'new';
+			};
 			// If there are any animations, want for the animationend event.
-			addEventListener('animationend', swap, { once: true });
+			addEventListener('animationend', fallbackSwap, { once: true });
 			// If there are no animations, go ahead and swap on next tick
 			// This is necessary because we do not know if there are animations.
 			// The setTimeout is a fallback in case there are none.
-			setTimeout(() => !isAnimating && swap());
+			var timeout = setTimeout(() => !isAnimating && fallbackSwap());
```

Comment:
This is a valid use of `var`!

Reply:
We could also hoist up a `let` 😄 `var` would bleed out of this if block.

Reply:
I also think that hoisting a `let` is more explicit. If it's impossible, we should leave a comment explaining why `var` needs to be used. E.g.: [like TS does](https://github.com/microsoft/TypeScript/blob/83f3abda153f0f567e296a3c75cdca37c7eb065d/src/compiler/sourcemap.ts#L38)

Reply:
Oh my I totally forgot that a let does hoist too 🤦‍♂️

Reply:
Ok fixed. Let's never speak of this again.